src/csharp/Microsoft.Spark.Worker.UnitTest/CommandExecutorTests.cs

```diff
@@ -601,10 +594,10 @@ Int64Array ConvertInt64s(Int64Array int64s)
                     Assert.Equal($"udf: {i}", stringArray.GetString(i));
                 }
 
-                var longArray = (Int64Array)outputBatch.Column(1);
+                var doubleArray = (DoubleArray)outputBatch.Column(1);
```

Comment:
long -> double occurs in the line 523 above. `dataFrame.Column(1) + 100` returns a double column.

Reply:
That seems wrong. `PrimitiveColumn<long>` + `int` should return a `long`.  Just like how:

```C#
long x = 0;
var y = x + 5;

y.GetType()
```

returns `System.Int64`.

---
In reply to: [330316160](https://github.com/dotnet/spark/pull/277#discussion_r330316160) [](ancestors = 330316160)

Reply:
Would you consider this something we HAVE to change? For example, `long` + `float` results in a `float` because the compiler knows to make the conversions. The DataFrame doesn't attempt to convert because each column's `Memory` has to be cloned in the conversion. For a chain of operations, this seems wasteful? Instead it converts once to a `PrimitiveColumn<double>`(or `PrimitiveColumn<decimal>`) when there is a type mismatch so all subsequent operations can work without cloning. 

The other alternative I'd considered was converting only when the underlying `Memory` had to be changed. `PrimitiveColumn<long>` + `int` would involve no conversions for example. I didn't think we should duplicate all the conversion logic in the compiler however, so I defaulted to what we have now. Do you know of an easier/better way to do this?

Reply:
I find it very surprising that adding, subtracting, or multiplying two integers would result in a floating-point number. IMO, I think that is something that should be changed.

But I'm not all customers, so maybe getting more thoughts/opinions here would be a good idea. But matching normal C# behavior seems like it is reasonable default behavior.